### PR TITLE
Add installation script to install CLI

### PR DESCRIPTION
**Pull request by JoelClemence**
* Add readme documentation on how to install CLI

### CODE REVIEW

File: install.sh [hunk 1]
```diff
@@ -0,0 +1,408 @@
...
+
+# Logs error and then exits process with error exit code
+panic() {
+  log ERROR "${1:?message required}" >&2
+  exit 1
+}
+
+# Logs messages to terminal and controls logging
+log() {
+  log_level="${1?:level required}"
```
Comment (sbasile-ch) [inline install.sh:51]: Shouldn't it be 
```
   log_level="${1:?level required}"
```
like below `$2` ?

File: install.sh [hunk 2]
```diff
@@ -0,0 +1,408 @@
...
+    jq -r '.[] | .name' "${releases_file}"
+    printf -- '%s' "$?" >"${temp_d}"/jq_status
+  } |
+    {
+      grep -q "${VERSION}"
+      printf -- '%s' "$?" >"${temp_d}"/grep_status
+    }
+
+  if ! grep -q '^0$' "${temp_d}"/jq_status ||
+    ! grep -q '^0$' "${temp_d}"/grep_status; then
```
Comment (sbasile-ch) [inline install.sh:102]: 1 grep should suffice

```
! grep -q '^0$' "${temp_d}"/jq_status "${temp_d}"/grep_status

```

File: install.sh [hunk 3]
```diff
@@ -0,0 +1,408 @@
...
+# operating system is unsupported by the CLI (i.e. missing compiled CLI
+# artifacts). Panics if not OSX or Linux compatible
+determine_operating_system() {
+  log DEBUG 'determining operating system'
+  uname_out="$(uname -a)"
+  case "${uname_out}" in
+  *Microsoft*) OS="linux" ;; #must be first since Windows subsystem for linux will have Linux in the name too
+  *microsoft*) OS="linux" ;; #WARNING: My v2 uses ubuntu 20.4 at the moment slightly different name may not always work
+  Linux*) OS="linux" ;;
+  Darwin*) OS="darwin" ;; # Apple Mac
```
Comment (sbasile-ch) [inline install.sh:118]: Making it case insensitive (i.e. lowercasing) the 2 Microsoft will be 1:
```
uname_out="$(uname -a | tr '[:upper:]' '[:lower:]')"
```

File: install.sh [hunk 4]
```diff
@@ -0,0 +1,408 @@
...
+    panic "Unsupported operating system. Must be Linux or Mac."
+    ;;
+  esac
+}
+
+# Sets the variable CHIPSET with the name of the current chipset architecture.
+# Panics if the chipset is not arm64 or x86
+determine_chipset() {
+  log DEBUG 'determining chipset'
+  uname_out="$(uname -m)"
```
Comment (sbasile-ch) [inline install.sh:129]: Same. Lowercase to keep it independent

File: install.sh (hunk 3 above)
Comment (JoelClemence) [reply install.sh:118]: oddly this is interfering with the output

> `Dagwin ABM-WVG6010RD0 23.4.0 Dagwin Kfgnfl Vfgsion 23.4.0b Fgi Mag 15 00b10b42 PDT 2024; gootbxnc-10063.101.17~1/RELEASE_ARM64_T6000 agm64`

File: install.sh [hunk 5]
```diff
@@ -0,0 +1,408 @@
...
+  install_version="$(jq -r '.name' "${release_file}")"
+  log INFO "Installing version ${install_version}"
+
+  jq -r --arg expected_suffix "-${OS}-${CHIPSET}.tar.gz" \
+    '.assets[] | select(.name | endswith($expected_suffix)) | .browser_download_url' \
+    "${release_file}" >"${temp_d}"/download_url
+}
+
+# Downloads the tarball to temporary directory
+download_cli_tarball() {
```
Comment (ch-tstrickland) [inline install.sh:174]: This will not, by itself, detect a failure to download. A 404 will be missed, as curl ran correctly.

Comment (JoelClemence) [reply install.sh:172]: Fixed with `--force-with-body`

File: install.sh [hunk 6]
```diff
@@ -0,0 +1,406 @@
...
+}
+
+# Sets the variable OS to the name of the operating system. Panics if the
+# operating system is unsupported by the CLI (i.e. missing compiled CLI
+# artifacts). Panics if not OSX or Linux compatible
+determine_operating_system() {
+  log DEBUG 'determining operating system'
+  uname_out="$(uname -a | tr '[:upper:]' '[:lower:]')"
+  case "${uname_out}" in
+  *microsoft*) OS="linux" ;; #WARNING: My v2 uses ubuntu 20.4 at the moment slightly different name may not always work
```
Comment (ch-tstrickland) [inline install.sh:114]: lowercasing Microsoft fails to distinguish between WSL & WSL2. Also, why not call it wsl2? It's not strictly linux. You could nick the existing function from setup_lib.bash.

Comment (JoelClemence) [reply install.sh:114]: It is all about working out which binary to download options are `linux` or `darwin`

Comment (JoelClemence) [reply install.sh:114]: For context take a look at: https://github.com/companieshouse/chs-dev/releases/tag/0.1.4 at the available tarballs

File: install.sh (hunk 5 above)
Comment (ch-tstrickland) [reply install.sh:172]: There is a solution to this issue, in the current incarnation of download_file in setup_lib.bash.

Comment (ch-tstrickland) [reply install.sh:172]: Ignore my last comment, your solution looks fine.

**Review by sbasile-ch — APPROVED** [review]
LGTM

File: install.sh (hunk 6 above)
Comment (JoelClemence) [reply install.sh]: Therefore I don't think there is the need here to distinguish between the two unless they need different binaries?

**Review by ch-tstrickland — APPROVED** [review]

**Pull request merged**